server/types/socket.ts

```diff
@@ -0,0 +1,19 @@
+import type { RoomSchema } from "../models/room.model.ts";
+
+export interface ConnectEvent {
+	event: "Connected";
+	userId: string;
+}
+
+export interface RoomUpdateEvent {
+	event: "RoomUpdate";
+	roomData: RoomSchema;
+}
+
+export interface JoinEvent {
```

Comment:
Is it worth having a separate event type for joining? The amount of data being sent for a full `RoomUpdateEvent` is trivial, so why not send it every time so we have less complexity?

Reply:
Ignore that comment, I misunderstood the implementation.